### ISSUE TITLE
Feature request: Export label informatin of selected points in a projection

### ISSUE THREAD

**Issue by benjaminengel** [tensorflow/tensorboard#820]
Hi there, 
I would like to export the labels of my selected points in a projection, in order to import them and retrain my model on these points.

One way could be to append the information on a bookmark. But I'm not where the label information are held, it would be great if you could give me some hints.

Regards

**Comment by jart** [comment]
I invite @nsthorat @francoisluus to comment on this feature request.

**Comment by francoisluus** [comment]
To immediately accomplish this make the selection, 'Edit by' with an identifiable label, and download the metadata that you then simply filter on when loading into your training dataset. You may need to add an extra metadata column beforehand which can contain your new labels that essentially say "use" or "don't use", where you filter out the "don't use" when creating your training array. I hope I am understanding your query correctly.

**Comment by benjaminengel** [comment]
Hi @francoisluus, thanks for the reply!

Yes this was exactly what I ment.

I've found the 'Edit by' (in version 0.4.0rc3 called 'Label by') and was hoping to see the label in a bookmark file after selecting them. But they don't show up. 

What I noticed in the meanwhile is that the bookmark contains a "selectedPoints" section which preserves the order of the input, that allowed me to reselect the points for model selection. 

Wouldn't it be a better option to just put the selected labels there as well? And if so, where could I start tinkering?

**Comment by jart** [comment]
I'm going to go ahead and close this one out. Thank you for helping our friend @francoisluus. I'll also note that [StackOverflow #tensorboard](http://stackoverflow.com/questions/tagged/tensorboard) has the most active community of folks answering questions.

**Comment by francoisluus** [comment]
@benjaminengel So a lot of the bookmarking magic happens in tensorboard/plugins/projector/vz_projector/vz-projector.ts, where you can look _getCurrentState()_ and _loadState()_. The labels are contained in _dataSet.spriteAndMetadataInfo_.

I am not sure if embedding metadata into the bookmark/state will be supported, unless the size and scope can be logically constrained. The current idea is just to download the changed metadata and to reload it, and also to load the bookmark. Another approach is to automatically include a 'selected' column in the downloaded/updated metadata file, a preference which the user can pre-specify with a tick box.
